rflx/pyrflx/typevalue.py

```diff
@@ -1077,24 +1099,26 @@ def __update_simplified_mapping(self, field: Optional[Field] = None) -> None:
             self._simplified_mapping[field.name_length] = field.typeval.size
             self._simplified_mapping[field.name_first] = field.first
             self._simplified_mapping[field.name_last] = field.last
+            self._simplified_mapping[self.message_last_name] = field.last
             return
 
         self._simplified_mapping = {}
         for v in self._fields.values():
-            if not v.set:
-                continue
-            if isinstance(v.typeval, ScalarValue):
+            if isinstance(v.typeval, ScalarValue) and v.set:
                 self._simplified_mapping[v.name_variable] = v.typeval.expr
-            self._simplified_mapping[v.name_length] = v.typeval.size
-            self._simplified_mapping[v.name_first] = v.first
-            self._simplified_mapping[v.name_last] = v.last
+            if isinstance(v.typeval, ScalarValue) or v.set:
+                self._simplified_mapping[v.name_length] = v.typeval.size
+            if isinstance(v.first, Number):
+                self._simplified_mapping[v.name_first] = v.first
+            if isinstance(v.last, Number):
+                self._simplified_mapping[v.name_last] = v.last
 
         # ISSUE: Componolit/RecordFlux#240
         self._simplified_mapping.update({ValidChecksum(f): TRUE for f in self._checksums})
```

Comment:
I removed this specific line (as it is unnecessary) and updated the issue in `_update_simplified_mapping`. The main problem is an indirect circular dependency between `ValidChecksum` and `_is_checksum_settable` I'd propose to fix this in #422 since it doesn't seem to be easily fixable and this part of the code needs to be refactored anyway.

Reply:
I'm fine with that. This should be documented in #422.

Reply:
I updated #422.